### DISCUSSION TITLE
Long term status of EnvKey as open source project

### DISCUSSION THREAD

**Discussion by drcongo** [envkey/envkey#114] [Q&A] [answered]
Hey Dane, firstly, thank you for EnvKey and all the time and effort you've put into it over the years - must have been a tough decision to wind down EnvKey cloud. Since the announcement, I've been trying out the alternatives, and found all of them lacking compared to EnvKey - combine the features of all of them and you would get close, but EnvKey really is the exemplar in this space for our needs. So with that in mind, I started thinking about self-hosting EnvKey as our replacement - the docs suggest this should be relatively easy to set up, but it does open up questions about long term viability. Do you think there could be enough of a community of enthusiastic users to keep it viable?

**Comment by benpsnyder** [comment]
@danenania my company is interested in self-hosting and we use Keycloak integrated with Azure Entra and want to tie together with OIDC. Infisical is cool but I was gravitating to envkey for our first "go" at this. Would you accept github sponsorships and perhaps pivot your financial model, moving all functionality to fully open-source?

**Comment by Kjue** [comment]
@danenania I'll reflect on OPs points too. Thank you for the great project! How do you see the longevity of the open source project? We are just in the process of setting things up and this comes up as a question.

**Reply by drcongo** [reply]
@Kjue - if it helps at all, I tried setting this up as self-hosted but was unable to authenticate against my instance in the EnvKey app. If you manage to get things working I'd appreciate any tips. I've tried pretty much every alternative on the market now and none of them even come close to EnvKey.

**Reply by henrywood** [reply]
@danenania Is there a guide / documentation somewhere that describes exactly how to set up and self-host Envkey ? What are the requirements ?

Thanks in advance !

**Comment by Matic** [comment]
Does anyone know if self hosted Envkey will keep working after the cloud versions shuts down? I'm thinking about the desktop client and so on.

**Reply by henrywood** [reply]
Is this thread dead ?
I would still like to know how to self-host Envkey ?
Is there some documentation or information on that subject somewhere @danenania ?

**Comment by danenania** [comment]
Hi @henrywood here you go! https://docs-v2.envkey.com/docs/self-hosting-open-source-envkey

**Comment by danenania** [comment, marked as answer]
Sorry everyone! I've unfortunately just been extremely pressed for time (and frankly, overwhelmed) with this whole transition and a new project that I'm trying to get started.

I will write up a better answer soon, but for now I'll just say that I would love to keep the open source project going... and that is my intention.

There is another question about open sourcing the business self-hosted and cloud infrastructure. I'm open to that as well and am considering it. 

That said, I unfortunately *really* don't have time to act as the maintainer. So someone (or better, some people) would need to be willing to fully take it over.

Just to be totally transparent, I want to say that one of the reasons I decided to move on from EnvKey is that it has some very rare but potentially nasty concurrency and key management race conditions buried in its innards. At EnvKey's peak scale, they would sometimes not show themselves for several months (doing ~100M requests per month), and then sometimes there would be 10 in a week. They are not security issues, but can cause data corruption and in the worst cases, data loss.

In working on my new project's (https://plandex.ai) v2, I recently went on an extended hours long deep dive on consistency, serialization, distributed locks, and associated error handling, and with the help of OpenAI's O1-pro model I was able to effectively remove all race conditions and locking problems (which Plandex  v1 has also had occasional issues with), and now for the first time in my life, I think I *might* have just built something that can scale without occasional race conditions.

All that to say, these issues are likely fixable by someone who has sufficient motivation and an o1-pro subscription.

So yeah, if anyone wants to take over as a maintainer, then please feel free to raise your hand. I'd say that since stability is one of EnvKey's most important features, and it already has quite a lot of features (and complexity), my suggestion would be to focus on bug fixes and tests vs. adding new features.

**Comment by VanCoding** [comment]
@danenania You've mentioned in your winding message that Infisical also has end-to-end encryption. Is this really true? I was in the process of evaluating Vault but then read the part abound E2E-encryption of EnvKey and found it to be a very good selling point. I'd really like the decryption of secrets to happen on the client. But after looking at Infisical I'm not really sure that this is realy the case there.
